drivers/npmx/npmx_shell.c

```diff
@@ -526,457 +547,466 @@ static int cmd_adc_meas_vbat_get(const struct shell *shell, size_t argc, char **
+	default:
+		shell_error(shell, "Error: Wrong vout select:");
+		print_help(shell, 0, "Vset pin");
+		print_help(shell, 1, "Software");
+		return 0;
+	}
 
-	if (err != 0) {
-		shell_error(shell, "Error: voltage value has to be an integer.");
+	npmx_error_t err_code = npmx_buck_vout_select_set(buck_instance, vout_select);
+	if (!check_error_code(shell, err_code)) {
+		print_set_error(shell, "vout select");
 		return 0;
 	}
 
-	npmx_charger_voltage_t voltage = npmx_charger_voltage_convert(volt_set);
+	print_usuccess(shell, select);
+	return 0;
+}
 
-	if (voltage == NPMX_CHARGER_VOLTAGE_INVALID) {
-		shell_error(shell, "Error: wrong termination voltage value.");
+static int buck_voltage_get(const struct shell *shell, size_t argc, char **argv,
+			    npmx_error_t (*voltage_get)(npmx_buck_t const *, npmx_buck_voltage_t *))
+{
+	args_info_t args_info = { .expected_args = 1,
+				  .arg = {
+					  [0] = { SHELL_ARG_TYPE_UINT_INDEX, "buck" },
+				  } };
+	if (!arguments_check(shell, argc, argv, &args_info)) {
 		return 0;
 	}
 
-	npmx_charger_t *charger_instance = npmx_charger_get(npmx_instance, 0);
-	uint32_t modules_mask;
-	npmx_error_t err_code = npmx_charger_module_get(charger_instance, &modules_mask);
+	npmx_buck_t *buck_instance = buck_instance_get(shell, args_info.arg[0].result.uvalue);
+	if (buck_instance == NULL) {
+		return 0;
+	}
+
+	npmx_error_t err_code;
+	npmx_buck_voltage_t buck_voltage;
+	if (voltage_get == npmx_buck_normal_voltage_get) {
+		npmx_buck_vout_select_t vout_select;
+		err_code = npmx_buck_vout_select_get(buck_instance, &vout_select);
+		if (!check_error_code(shell, err_code)) {
+			print_get_error(shell, "vout select");
+			return 0;
+		}
+
+		if (vout_select == NPMX_BUCK_VOUT_SELECT_VSET_PIN) {
+			err_code = npmx_buck_status_voltage_get(buck_instance, &buck_voltage);
+		} else if (vout_select == NPMX_BUCK_VOUT_SELECT_SOFTWARE) {
+			err_code = voltage_get(buck_instance, &buck_voltage);
+		} else {
+			shell_error(shell, "Error: invalid vout select.");
+			return 0;
+		}
+	} else {
+		err_code = voltage_get(buck_instance, &buck_voltage);
+	}
 
 	if (!check_error_code(shell, err_code)) {
-		shell_error(shell, "Error: unable to get charger module status.");
+		print_get_error(shell, "buck voltage");
 		return 0;
 	}
 
-	if ((modules_mask & NPMX_CHARGER_MODULE_CHARGER_MASK) != 0) {
-		shell_error(shell,
-			    "Error: charger must be disabled to set battery termination voltage.");
+	uint32_t voltage_mv;
+	if (!npmx_buck_voltage_convert_to_mv(buck_voltage, &voltage_mv)) {
+		print_convert_error(shell, "buck voltage", "millivolts");
 		return 0;
 	}
 
-	err_code = func(charger_instance, voltage);
+	print_uvalue_unit(shell, voltage_mv, "mV");
+	return 0;
+}
 
-	if (check_error_code(shell, err_code)) {
-		shell_print(shell, "Success: %s mV.", argv[1]);
-	} else {
-		shell_error(shell, "Error: unable to set battery termination voltage value.");
+static int buck_voltage_set(const struct shell *shell, size_t argc, char **argv,
```

Comment:
According to your suggestion to the guide [Writing npmx shell](https://nordicsemi.atlassian.net/wiki/spaces/NRFX/pages/301893654/Writing+npmx+shell), the setter should be first. Also in other places.